### ISSUE TITLE
`test_posix_fallocate()` fails under `test_posix` under WASI

### ISSUE THREAD

**Issue by brettcannon** [python/cpython#111804]
```
======================================================================
ERROR: test_posix_fallocate (__main__.PosixTester.test_posix_fallocate)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Lib/test/test_posix.py", line 405, in test_posix_fallocate
    posix.posix_fallocate(fd, 0, 10)
OSError: [Errno 58] Not supported

----------------------------------------------------------------------
```

Detected with WASI-SDK 20 and wasmtime 14.

### Linked PRs
* gh-111869
* gh-111919
* gh-111920

**Comment by brettcannon** [comment]
I have asked at https://bytecodealliance.zulipchat.com/#narrow/stream/219900-wasi/topic/Who.20is.20expected.20to.20make.20.60posix_fallocate.60.20work.3F to find out who I should reach out to for finding out what this function errors out.

**Comment by brettcannon** [comment]
Turns out[ `posix_fallocate()` was in WASI preview1, but has been yanked in preview2 due to inconsistent semantics across OSs](https://bytecodealliance.zulipchat.com/#narrow/stream/219900-wasi/topic/Who.20is.20expected.20to.20make.20.60posix_fallocate.60.20work.3F/near/400666781). Best solution then is to just skip `posix_fallocate()` on WASI unconditionally.

**Comment by brettcannon** [comment]
Fixed in `main`, but the backports to 3.11 and 3.12 still need to be done.
